cloud/ec2_util.go

```diff
@@ -282,40 +282,25 @@ func validateUserDataSize(userData, distroID string) error {
 	return errors.WithStack(err)
 }
 
+// hostInstancePair represents a host and its associated EC2 instance.
+type hostInstancePair struct {
+	host     *host.Host
+	instance *types.Instance
+}
+
 // cacheHostData caches host information received from AWS about the instance.
-func cacheHostData(ctx context.Context, env evergreen.Environment, h *host.Host, instance *types.Instance, client AWSClient) error {
-	if instance.Placement == nil || instance.Placement.AvailabilityZone == nil {
-		return errors.New("instance missing availability zone")
-	}
-	if instance.LaunchTime == nil {
-		return errors.New("instance missing launch time")
-	}
-	if instance.PublicDnsName == nil {
-		return errors.New("instance missing public DNS name")
-	}
-	if instance.PrivateIpAddress == nil {
-		return errors.New("instance missing private IP address")
+func cacheHostData(ctx context.Context, env evergreen.Environment, client AWSClient, pair hostInstancePair) error {
```

Comment:
I understand the DB optimizations between single vs multi insert are why this and `cacheAllHostData` are distinct, but it seems some of the logic between these two can be shared